src/GameState/Board.tsx

```diff
@@ -56,45 +57,102 @@ export const Board = (props: any) => {
           height: '100%',
         }}
       >
-        {boardCoordinate.map((item) => (
-          <TranslateView
-            activated={marbleIsAnimated(
-              item,
-              animatedMarble?.marblesCoordinate,
-            )}
-            direction={animatedMarble?.direction}
-            style={{
-              position: 'absolute',
-              left: `${item.x * boxMarbleSize}%`,
-              top: `${item.y * boxMarbleSize}%`,
-              width: boxMarbleSizePourcent,
-              height: boxMarbleSizePourcent,
-              alignItems: 'center',
-              justifyContent: 'center',
-            }}
-            key={item.x + ',' + item.y}
-          >
-            <Box
-              bg={
-                marbleClickedCoordinates?.x === item.x &&
-                marbleClickedCoordinates?.y === item.y
-                  ? BOARD_GREEN
-                  : ''
-              }
-            >
-              {item.value != 0 && (
-                <Marble
-                  value={item.value}
-                  size={marbleColorSize}
-                  rowIndex={item.y}
-                  cellIndex={item.x}
-                  setMarbleClickedCoordinates={setMarbleClickedCoordinates}
-                  setMarbleClicked={props.setMarbleClicked}
-                />
+        {boardCoordinate.map((item) => {
+          const zIndex =
+            marbleClickedCoordinates?.x === item.x &&
+            marbleClickedCoordinates?.y === item.y
+              ? 1
+              : 0;
+          return (
+            <TranslateView
+              activated={marbleIsAnimated(
+                item,
+                animatedMarble?.marblesCoordinate,
```

Comment:
The content is bigger and bigger. It is time to extract it in a component.

Reply:
As we need to deploy a working version of the app for this afternoon demo, we will take care of your comments and our corrections in another branch.